### PR TITLE
Add ynh_check_starting

### PR DESCRIPTION
**Pull request by maniackcrudelis**
## Problem
- *504 error on the current testing*

## Solution
- *Add ynh_check_starting to be sure searx has started before testing its ur*

## PR Status
Work finished. Package_check, basic tests and upgrade from last version OK.  
Could be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [x] **Code review** : JimboJoe
- [x] **Approval (LGTM)** : Josué
- [x] **Approval (LGTM)** : JimboJoe
- **CI succeeded** : [Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/searx_ynh%20add_ynh_check_starting%20(Official)/)  
When the PR is mark as ready to merge, you have to wait for 3 days before really merge it.

### CODE REVIEW

File: scripts/restore [hunk 1]
```diff
@@ -13,12 +13,12 @@ set -eu
...
-# 	# Rapatrie le fichier de fonctions si il n'est pas dans le dossier courant
-# 	cp ../settings/scripts/_common.sh ./_common.sh
-# 	chmod a+rx _common.sh
-# fi
-# source _common.sh
+if [ ! -e _common.sh ]; then
+	# Rapatrie le fichier de fonctions si il n'est pas dans le dossier courant
+	cp ../settings/scripts/_common.sh ./_common.sh
+	chmod a+rx _common.sh
+fi
```
Comment (Josue-T) [inline scripts/restore]: Why do you can't do something like that : `source ../settings/scripts/_common.sh`

I did this in all of my app without problem.

**Comment by JimboJoe** [comment]
Still 504 gateway timeout errors in CI...

File: scripts/restore (hunk 1 above)
Comment (maniackcrudelis) [reply scripts/restore]: It's an old way to to that, it was probably because we needed a chmod +x

**Comment by maniackcrudelis** [comment]
So... that means that we have to debug that directly on this CI...

File: scripts/restore (hunk 1 above)
Comment (Josue-T) [reply scripts/restore]: So, could we remove it to make the code more clean ?

Comment (maniackcrudelis) [reply scripts/restore]: If it works for you, yes we can do that

**Comment by maniackcrudelis** [comment]
Well... It still doesn't work...
But, I think that I found the error here
```
Found existing installation: pyOpenSSL 0.14
Not uninstalling pyOpenSSL at /usr/lib/python2.7/dist-packages, outside environment /opt/yunohost/searx
```
...Shit, I just saw that it says "Not UNinstalling"...
Why this fucking python's still saying that "The pyopenssl package has to be installed" since it's quite obvious that pyopenssl IS ALREADY INSTALLED !!!

That's going to be all for today... Any help would be welcome...

**Comment by maniackcrudelis** [comment]
Still broken...
If you want to try to work on that, I'm working in `/home/yunohost.app/ssh_chroot_directories/Official/data/searx/searx_ynh add_ynh_check_starting` on ci-apps-dev.yunohost.org.

The error is around the line 97.
[Next test](https://ci-apps-dev.yunohost.org/jenkins/view/Official/job/searx_ynh%20add_ynh_check_starting%20(Official)/12/consoleFull) will be with:
```
set +u; source $final_path/bin/activate; set -u
pip install -U setuptools
pip install pyopenssl
pip install --upgrade --requirement $final_path/requirements-ynh.txt
```
`pip install pyopenssl` is new in this test.

**Comment by maniackcrudelis** [comment]
Any help on this issue would be welcome. I have no idea why python is shitting here...

**Comment by maniackcrudelis** [comment]
Last test with
```
#virtualenv --system-site-packages "$final_path"
virtualenv "$final_path"
set +u; source $final_path/bin/activate; set -u
pip install --upgrade setuptools
pip install --upgrade --requirement $final_path/requirements-ynh.txt
```
and
```
# Have a look to https://github.com/asciimoo/searx/blob/master/requirements.txt for each upgrade of Searx
certifi==2017.11.5
flask==0.12.2
flask-babel==0.11.2
idna==2.5
pygments==2.1.3
python-dateutil==2.6.1
pyyaml==3.12
requests[socks]==2.18.4

# Additionnals requirements
markupsafe>=0.23
ndg-httpsclient>=0.4.2
pyasn1>=0.1.9
pyasn1-modules>=0.0.8
cffi>=1.6

PySocks>=1.6.5
pyOpenSSL>=16.2.0
lxml>=3.7.1
cryptography>=1.7.1
Babel>=2.3.4
Werkzeug>=0.12.1
asn1crypto>=0.22.0
pyparsing>=2.2.0
```

**Comment by maniackcrudelis** [comment]
Ok... I eventually removed the https redirection from the nginx conf, in order to be sure that pyopenssl is the problem.
It appears that there's still a timeout, even without https, so the issue is elsewhere...
But where... ? There's nothing in the nginx log, only a timeout, that I already known.
Nothing neither in the UWSGI log...

**Comment by maniackcrudelis** [comment]
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/searx_ynh%20add_ynh_check_starting%20(Official)/)  !!! \o/

Ready to be reviewed !!!

**Pull request merged**